src/python/pants/core/target_types.py

```diff
@@ -206,5 +216,142 @@ class GenericTarget(Target):
     core_fields = (*COMMON_TARGET_FIELDS, Dependencies)
 
 
+# -----------------------------------------------------------------------------------------------
+# `archive` target
+# -----------------------------------------------------------------------------------------------
+
+# TODO(#10888): Teach project introspection goals that this is a special type of the `Dependencies`
+#  field.
+class ArchivePackages(StringSequenceField):
+    """Addresses to any targets that can be built with `./pants package`.
+
+    Pants will build the assets as if you had run `./pants package`. It will include the
+    results in your archive using the same name they would normally have, but without the `dist/`
+    prefix.
+
+    You can include anything that can be built by `./pants package`, e.g. a `python_binary`,
+    `python_awslambda`, or even another `archive`.
+    """
+
+    alias = "packages"
+
+
+# TODO(#10888): Teach project introspection goals that this is a special type of the `Dependencies`
+#  field.
+class ArchiveFiles(StringSequenceField):
+    """Addresses to any `files` or `relocated_files` targets to include in the archive, e.g.
+    `["resources:logo"]`.
+
+    This is useful to include any loose files, like data files, image assets, or config files.
+
+    This will ignore any targets that are not `files` or `relocated_files` targets. If you instead
+    want those files included in any packages specified in the `packages` field for this target,
+    then use a `resources` target and have the original package (e.g. the `python_library`)
+    depend on the resources.
+    """
+
+    alias = "files"
+
+
+class ArchiveFormatField(StringField):
+    """The type of archive file to be generated.
+
+    Note that `zip` files will not be compressed.
```

Comment:
I assume this means that files inside the zip will be compressed but that the entire archive will not be compressed (since zip compresses file-by-file)? If so, this comment should be clarified since it could be read to say that the files will be stored in the zip without any compression whatsoever.

Reply:
For the JVM we did not compress because `jvm_app` targets were guaranteed to be full of jar files, which were themselves compressed.

This heuristic _might_ still apply if the majority of the stuff inside the archive is already compressed (I think that PEX files are?) but it's maybe still an odd default.

Reply:
I think the more accurate term I was after is `deflated`: https://docs.python.org/3/library/zipfile.html#zipfile.ZIP_DEFLATED

I'm gonna remove this comment, I think. Probably causes more confusion than necessary. We do what the default of running `zip my_zip.zip f1 f2 f3` would do.